src/Cpp/1-getting-started/1-3-4-3DRendering/3DRenderingApplication.cpp

```diff
@@ -125,11 +131,62 @@ bool Rendering3DApplication::Initialize()
     }
 
     CreateSwapchainResources();
-
+    CreateRasterState();
+    CreateDepthStencilView();
+    CreateDepthState();
     CreateConstantBuffers();
     return true;
 }
 
+void Rendering3DApplication::CreateRasterState()
+{
+    D3D11_RASTERIZER_DESC rasterDesc{};
+    rasterDesc.CullMode = D3D11_CULL_NONE;
+    rasterDesc.FillMode = D3D11_FILL_SOLID;
+
+    _device->CreateRasterizerState(&rasterDesc, &_rasterState);
+}
+
+void Rendering3DApplication::CreateDepthStencilView()
+{
+    D3D11_TEXTURE2D_DESC texDesc{};
+    texDesc.Height = GetWindowHeight();
+    texDesc.Width = GetWindowWidth();
+    texDesc.ArraySize = 1;
+    texDesc.SampleDesc.Count = 1;
+    texDesc.MipLevels = 1;
+    texDesc.BindFlags = D3D11_BIND_DEPTH_STENCIL;
+    texDesc.Format = DXGI_FORMAT_R32_TYPELESS;
+
+    ID3D11Texture2D* texture = nullptr;
+    if (FAILED(_device->CreateTexture2D(&texDesc, nullptr, &texture)))
+    {
+        std::cout << "DXGI: Failed to create texture for DepthStencilView\n";
```

Comment:
Heh `std::cout` also triggers me, but once all the tuts are in place, we shall replace with `spdlog` for some more real worldyisms